src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -353,3 +368,26 @@ class LiteralCoercionSegment(BaseSegment):
         OneOf("DATE", "DATETIME", "TIME", "TIMESTAMP"),
         Ref("QuotedLiteralSegment"),
     )
+
+
+@bigquery_dialect.segment()
```

Comment:
Hurray for the new dialect inheritance support! This was pretty straightforward and avoids a lot of copy/paste.